haskell-tests/Types/CBOR.hs

```diff
@@ -295,17 +313,101 @@ testEncodedTokenOperations = describe "EncodedTokenOperations JSON serialization
                     invalidEncTops1
             )
 
+testTokenMetadataUrlJSON :: Spec
+testTokenMetadataUrlJSON = describe "TokenMetadataUrl JSON serialization" $ do
+    let tmUrl =
+            TokenMetadataUrl
+                { tmUrl = "https://example.com/token-metadata",
+                  tmChecksumSha256 = Just $ SHA256.Hash (FBS.pack $ replicate 32 0xab),
+                  tmAdditional = Map.empty
+                }
+    it "Serialize/Deserialize roundtrip success" $
+        assertEqual
+            "Deserialized"
+            (Just tmUrl)
+            ( AE.decode $
+                AE.encode
+                    tmUrl
+            )
+    it "Serializes to expected JSON object" $
+        case AE.toJSON tmUrl of
+            AE.Object o -> assertBool "Does not contain field url" $ AE.member "url" o
+            _ -> assertFailure "Does not encode to JSON object"
+
+testTokenMetadataUrlCBOR :: Spec
+testTokenMetadataUrlCBOR = describe "TokenMetadataUrl CBOR serialization" $ do
+    it "Decodes TokenMetadataUrl without additional" $
+        assertEqual
+            "Decoded CBOR"
+            ( Right
+                ( "",
+                  TokenMetadataUrl
+                    { tmUrl = "https://abc.token/meta",
+                      tmChecksumSha256 = Just $ SHA256.Hash (FBS.pack $ replicate 32 0xab),
+                      tmAdditional = Map.empty
+                    }
+                )
+            )
+            ( deserialiseFromBytes
+                decodeTokenMetadataUrl
+                "\xA2\x63\x75\x72\x6C\x76\x68\x74\x74\x70\x73\x3A\x2F\x2F\x61\x62\x63\x2E\x74\x6F\x6B\x65\x6E\x2F\x6D\x65\x74\x61\x6E\x63\x68\x65\x63\x6B\x73\x75\x6D\x53\x68\x61\x32\x35\x36\x58\x20\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB"
```

Comment:
I got weird mismatches when I tried to represent this as a string as done in the other tests. 

`"\x6EchecksumSha256"` would get the hex representation `"ec6865636b73756d5368613235365"` which is incorrect (should be `"6e636865636b73756d5368613235365`".

To avoid this I use the raw byte representation instead.

This goes for the other tests added as well.

Reply:
I also recommend splitting this across a few lines.